modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/PersistentPageMemory.java

```diff
@@ -1029,6 +1037,28 @@ public long loadedPages() {
         return total;
     }
 
+    /**
+     * Returns the count of dirty pages across all segments.
+     */
+    public long dirtyPagesCount() {
+        Segment[] segments = this.segments;
+        if (segments == null) {
+            return 0;
+        }
+
+        long total = 0;
+
+        for (Segment seg : segments) {
+            if (seg == null) {
```

Comment:
I guess this is a copy-paste. Looking at other places, we never have this null check anywhere except for `loadedPages()`, so I guess we don't need it here either. Let's remove it from `loadedPages()` as well

Reply:
It's funny, but it's the other way around: the original position (in the middle of the construct) means that the reference to array can be null itself, but if you put the annotation before the construct, it means that array elements are nullable. See https://stackoverflow.com/a/50277992/7637120 and note that `@Nullable` is meta-annotated with `@Target(ElementType.TYPE_USE)`